### ISSUE TITLE
proposal: Another set of renames

### ISSUE THREAD

**Issue by filiph** [alnitak/flutter_soloud#52]
**Description**

I have another set of renames that I'd like to run by you (when you have time!). Basically, I'm reading the SoLoud page now (e.g. https://solhsa.com/soloud/concepts.html) and I think we might want to stick to their terminology. 

![](https://solhsa.com/soloud/images/concepts.png)

For example, `SoundProps` might become `AudioSource`. Possibly `SoLoud.initialize()` to `SoLoud.init()`, and `shutdown()` to `deinit()`, although here I'm uncertain since those are against Dart style. But looking at the C# wrapper, for example, I can see that they also break C# conventions (https://solhsa.com/soloud/cs_api.html). Or maybe `deinit()` should be a synchronous "kill switch" for when there's no time at the very end of the app/game's lifecycle — and we keep `shutdown()` for a softer, more thorough exit.

Apart from renaming, I'd love to move the documentation as close to the original (C++) library as possible.

**Comment by filiph** [comment]
Possibly `SoundHandle` should be `Voice`? Although even the C++ docs are inconsistent and call it mostly "sound handle".

**Comment by filiph** [comment]
cc @alnitak for you thoughts on this

**Comment by alnitak** [comment]
> Or maybe deinit() should be a synchronous "kill switch"

I just came across this situation with #54 ! :) So I also think your `deinit()` "kill switch" is a good idea.

> For example, `SoundProps` might become `AudioSource`.

IIRC I named `SoundProps` (SoundProperties shortened), to store the ways to manage it in [native code](https://github.com/alnitak/flutter_soloud/blob/e79a897a655ad6ac61d23b9b9d50b2e72b8d9c1e/src/player.h#L31).
Into `SoundProps` we could add for example the audio file name, like in native, but also any other properties we like. It's not  really like [`Soloud::AudioSource`](https://solhsa.com/soloud/audiosource.html) which in `SoundProps` could be represented by `SoundHash`.
Anyway, I like `AudioSource` name more than `SoundProps`!

> Possibly SoundHandle should be Voice? Although even the C++ docs are inconsistent and call it mostly "sound handle".

I recall that I get a bit confused too when I read `voice`. Maybe it's a deeper Soloud lib concept needed maybe if/when we will go deeper integrating the lib into the plugin.  The Soloud author says "The handle is an integer, and internally tracks the voice and sound id, as well as an "uniqueness" value.", so I think the "voice" could be thought as a sub-type of a handle.
